core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -374,4 +380,40 @@ void validate(JsonNode element) {
           element);
     }
   }
+
+  public static void writeIntegerArray(
+      String property, Collection<Integer> items, JsonGenerator gen) throws IOException {
```

Comment:
We usually accept `Iterable` rather than `Collection` in cases where we just need to iterate so that we don't need to allocate a new list/set and collect into it. That would simplify some calls:

```
      JsonUtil.writeStringArray(
          MANIFESTS,
          Iterables.transform(snapshot.allManifests(DUMMY_FILE_IO), ManifestFile::path),
          generator);```

Reply:
makes sense, updated